code/datums/stars.dm

```diff
@@ -227,14 +227,31 @@
 	var/primary_resource
 	var/is_primary = 0
 
+	var/datum/station_module/module
 
 /datum/space_station/New(var/datum/planet/P)
 	planet = P
 	SSstarmap.stations += src
 
-/datum/space_station/proc/generate()
+
+/datum/space_station/proc/generate(var/list/module_weights)
 	// TODO: Implement a more sophisticated way of generating station stocks.
```

Comment:
if i leave this comment here, someone will eventually go "hey this is shit i can do it better" and then it'll be done better. ~iterations~